tests/phpunit/integration/Modules/Analytics_4/Conversion_Reporting/Conversion_Reporting_Events_SyncTest.php

```diff
@@ -67,8 +77,47 @@ public function test_sync_detected_events( $detected_events, $report_rows ) {
 		$this->assertEquals( $detected_events, $this->settings->get()['detectedEvents'] );
 	}
 
+	/**
+	 * @dataProvider report_dimensions_with_new_events
+	 */
+	public function test_sync_newly_detected_events( $initially_saved_events, $detected_new_events, $report_rows ) {
+		$this->setup_fake_handler_and_analytics( $report_rows );
+
+		$event_check = $this->get_instance();
+		$this->settings->merge(
+			array(
+				'detectedEvents' => $initially_saved_events,
+			)
+		);
+		$event_check->sync_detected_events();
+
+		$transient_detected_events = $this->transients->get( 'googlesitekit_conversion_reporting_detected_events' );
```

Comment:
Once transient names are moved to the class constant, we can replace it's usage here and in other places in the tests to use it directly 
```suggestion
        $transient_detected_events = $this->transients->get( Conversion_Reporting_Events_Sync::DETECTED_EVENTS_TRANSIENT );
```